theano/tensor/nnet/tests/test_nnet.py

```diff
@@ -156,41 +173,52 @@ def test_softmax_with_bias_trace(self):
         assert check_stack_trace(f, ops_to_check='last')
 
     def test_infer_shape(self):
-        admat = matrix()
+        admat = tensor3()
         advec = vector()
-        admat_val = np.random.rand(3, 4).astype(config.floatX)
-        advec_val = np.random.rand(4).astype(config.floatX)
+        axis = np.random.randint(3)
+        admat_val = np.random.rand(3, 4, 5).astype(config.floatX)
+        advec_val = np.random.rand([3, 4, 5][axis]).astype(config.floatX)
         self._compile_and_check([admat, advec],
-                                [SoftmaxWithBias()(admat, advec)],
+                                [SoftmaxWithBias()(admat, advec, axis)],
                                 [admat_val, advec_val], SoftmaxWithBias)
 
 
 class T_LogSoftmax(utt.InferShapeTester):
 
     def test0(self):
+        axis = np.random.randint(3)
+
         def f(a):
-            return logsoftmax_op(a)[:, 0]
-        utt.verify_grad(f, [np.random.rand(3, 4)])
+            return logsoftmax_op(a, axis)[:, 0]
+        utt.verify_grad(f, [np.random.rand(3, 4, 5)])
 
     def test1(self):
+        axis = np.random.randint(3)
+
         def f(a):
-            return logsoftmax_op(a)[:, 1]
-        utt.verify_grad(f, [np.random.rand(3, 4)])
+            return logsoftmax_op(a, axis)[:, 1]
+        utt.verify_grad(f, [np.random.rand(3, 4, 5)])
 
     def test2(self):
+        axis = np.random.randint(3)
+
         def f(a):
-            return logsoftmax_op(a)[:, 2]
-        utt.verify_grad(f, [np.random.rand(3, 4)])
+            return logsoftmax_op(a, axis)[:, 2]
+        utt.verify_grad(f, [np.random.rand(3, 4, 5)])
```

Comment:
This is currently failing because of a bug in ```SoftmaxWithBias```. I notified @bordesf and explained the cause in the other PR.